pkg/devfile/adapters/docker/component/adapter.go

```diff
@@ -36,6 +68,29 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 		return errors.Wrap(err, "unable to create or update component")
 	}
 
+	containers := utils.GetComponentContainers(a.Client, a.ComponentName)
+	// Find at least one pod with the source volume mounted, error out if none can be found
+	containerID, err := getFirstContainerWithSourceVolume(containers)
+	if err != nil {
+		return errors.Wrapf(err, "error while retrieving container for component: %s", a.ComponentName)
+	}
+
+	// Get a sync adapter. Check if project files have changed and sync accordingly
+	syncAdapter := sync.New(a.AdapterContext, &a.Client)
+	// podName is set to empty string on docker
+	// podChanged is set to false, since docker volume is always present even if container goes down
+	isPushRequired, err := syncAdapter.SyncFiles(parameters, "", containerID, false, componentExists)
```

Comment:
Can you change `isPushRequired` to something else, since part of what `SyncDoes` is push the changes to the running container. Maybe something like `buildRequired` or `execRequired`?